src/catkin_pkg/packages.py

```diff
@@ -104,11 +115,12 @@ def find_packages_allowing_duplicates(basepath, exclude_paths=None, exclude_subs
     :param warnings: Print warnings if None or return them in the given list
     :returns: A dict mapping relative paths to ``Package`` objects ``dict``
     """
-    packages = {}
     package_paths = find_package_paths(basepath, exclude_paths=exclude_paths, exclude_subspaces=exclude_subspaces)
-    for path in package_paths:
-        packages[path] = parse_package(os.path.join(basepath, path), warnings=warnings)
-    return packages
+    parser = _PackageParser(capture_warnings=isinstance(warnings, list))
+    results = multiprocessing.Pool(4).map(parser, package_paths)
```

Comment:
Should we use something more "dynamic" like `multiprocessing.cpu_count()` as a default (and fallback to a hard coded number like `4`)?